Stack/Opc.Ua.Core/Types/Encoders/JsonEncoder.cs

```diff
@@ -1328,14 +1436,14 @@ public void WriteVariant(string fieldName, Variant value)
 
             try
             {
-
                 bool isNull = (value.TypeInfo == null || value.TypeInfo.BuiltInType == BuiltInType.Null || value.Value == null);
 
-                if (UseReversibleEncoding && !isNull)
+                if (!isNull && EncodingToUse != JsonEncodingType.NonReversible)
                 {
                     PushStructure(fieldName);
                     // encode enums as int32.
                     byte encodingByte = (byte)value.TypeInfo.BuiltInType;
```

Comment:
@randy-armstrong should this be done for Verbose encoding? It leads to a Variant with Int32 built in type which contains string encoded enums/numbers? The BuiltInType.Enumeration is just an internal  helper, so which type to use instead?

Reply:
I tried what was discussed yesterday in the WG meeting, to set the type to string for the Verbose encoding. but that leads to a failing test case where a DataValue with a Variant with an array of enums is encoded but the decoded Variant contains an array of strings. This breaks indempotent encoding. Imho to encode Int32 as strings is the better option. needs discussion.